libkbfs/key_bundle_cache.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	lru "github.com/hashicorp/golang-lru"
+)
+
+// KeyBundleCacheStandard is an LRU-based implementation of the KeyBundleCache interface.
+type KeyBundleCacheStandard struct {
+	lru *lru.Cache
+}
+
+var _ KeyBundleCache = (*KeyBundleCacheStandard)(nil)
+
+type keyBundleCacheKey struct {
+	tlf         TlfID
+	bundleIDStr string
+	isWriter    bool
+}
+
+// NewKeyBundleCacheStandard constructs a new KeyBundleCacheStandard with the given
+// cache capacity.
+func NewKeyBundleCacheStandard(capacity int) *KeyBundleCacheStandard {
+	head, err := lru.New(capacity)
+	if err != nil {
+		panic(err.Error())
+	}
+	return &KeyBundleCacheStandard{head}
+}
+
+// GetTLFReaderKeyBundle implements the KeyBundleCache interface for KeyBundleCacheStandard.
+func (k *KeyBundleCacheStandard) GetTLFReaderKeyBundle(
+	tlf TlfID, bundleID TLFReaderKeyBundleID) *TLFReaderKeyBundleV3 {
+	cacheKey := keyBundleCacheKey{tlf, bundleID.String(), false}
+	if entry, ok := k.lru.Get(cacheKey); ok {
+		if rkb, ok := entry.(*TLFReaderKeyBundleV3); ok {
+			return rkb
+		}
+		// Shouldn't be possible.
+		panic("Invalid key bundle type")
```

Comment:
it's possible that in the future the LRU cache will store entries other than V3 key bundles, right?